game.go

```diff
@@ -205,25 +216,34 @@ func (g *Game) currentTeam() Team {
 }
 
 func newGame(id string, state GameState) *Game {
-	rnd := rand.New(rand.NewSource(state.Seed))
+	// Reset seed and index if game has exhausted all words
+	if (state.PermIndex + wordsPerGame >= len(state.WordSet)) {
+		state = randomState(state.WordSet)
+	}
+
+	// used for generating words, consistent randomness across games
+	seedRnd := rand.New(rand.NewSource(state.Seed))
+	// used for all other random settings. starting team and shuffling
+	randRnd := rand.New(rand.NewSource(rand.Int63()))
+
 	game := &Game{
 		ID:           id,
 		CreatedAt:    time.Now(),
-		StartingTeam: Team(rnd.Intn(2)) + Red,
+		StartingTeam: Team(randRnd.Intn(2)) + Red,
 		Words:        make([]string, 0, wordsPerGame),
 		Layout:       make([]Team, 0, wordsPerGame),
-		GameState:    state,
+		GameState:    resetState(state),
 	}
 
-	// Pick 25 random words.
-	used := map[string]struct{}{}
-	for len(used) < wordsPerGame {
-		w := state.WordSet[rnd.Intn(len(state.WordSet))]
-		if _, ok := used[w]; !ok {
-			used[w] = struct{}{}
-			game.Words = append(game.Words, w)
-		}
+	// Pick the next `wordsPerGame` words from the
+	// randomly generated permutation
+	perm := seedRnd.Perm(len(state.WordSet))
+	permIndex := state.PermIndex
+	for _, i := range perm[permIndex:permIndex + wordsPerGame] {
+		w := state.WordSet[perm[i]]
+		game.Words = append(game.Words, w)
 	}
+	game.GameState.PermIndex = permIndex + wordsPerGame
```

Comment:
wdyt about advancing permIndex outside of newGame? That, combined with the change above, would ensure that you can reconstruct a game by calling `newGame(g.ID, g.GameState)`.

Reply:
I can make that happen. Since the shuffle and first team randomness needs to be unique across each game for a given id _but_ the seed is now consistent (until you hit all the words) -- let me look through the rand api and see if I can generate some deterministic randomness based off of seed + permIndex.

Reply:
done